bug: A large table is executed by ddl,the tianmu.log will print a large number of error.

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```sql
mysql> show create table ttt\G
*************************** 1. row ***************************
       Table: ttt
Create Table: CREATE TABLE `ttt` (
  `id` int NOT NULL,
  `name` varchar(10) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=TIANMU DEFAULT CHARSET=utf8mb4
1 row in set (0.00 sec)

mysql> select count(*) from ttt;
+----------+
| count(*) |
+----------+
|  5000000 |
+----------+
1 row in set (0.00 sec)

###tianmu.log
......
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152895
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152896
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152897
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152898
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152899
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152900
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152901
[ERROR] [combined_iterator.cpp:55] MSG: base pos: 152902
......
```

### Expected behavior

_No response_

### How To Reproduce

_No response_

### Environment

./mysqld  Ver 5.7.36-StoneDB-v1.0.3 for Linux on x86_64 (build-)
build information as follow:
        Repository address: https://github.com/stoneatom/stonedb:stonedb-5.7-dev
        Branch name: stonedb-5.7-dev
        Last commit ID: 758bce8a5
        Last commit time: Date:   Mon Apr 10 19:46:20 2023 +0800
        Build time: Date: Tue 11 Apr 2023 03:30:57 AM UTC

### Are you interested in submitting a PR to solve the problem?

- [X] Yes, I will!

--- a/storage/tianmu/executor/combined_iterator.cpp
+++ b/storage/tianmu/executor/combined_iterator.cpp
@@ -51,9 +51,6 @@ void CombinedIterator::Next() {
     }
   } else {
     base_iter_->Next();
-    if (base_iter_->Valid()) {
-      TIANMU_LOG(LogCtl_Level::ERROR, "base pos: %d", base_iter_->Position());
-    }
   }
 }
 
@@ -85,8 +82,8 @@ bool CombinedIterator::Valid() const { return Position() != -1; }
 bool CombinedIterator::IsBase() const { return is_base_; }
 
 bool CombinedIterator::BaseCurrentRowIsInvalid() const {
-  if (base_iter_->CurrentRowIsDeleted() || InDeltaDeletedRow.find(base_iter_->Position()) != InDeltaDeletedRow.end() ||
-      InDeltaUpdateRow.find(base_iter_->Position()) != InDeltaUpdateRow.end()) {
+  if (InDeltaDeletedRow.find(base_iter_->Position()) != InDeltaDeletedRow.end() ||
+      InDeltaUpdateRow.find(base_iter_->Position()) != InDeltaUpdateRow.end() || base_iter_->CurrentRowIsDeleted()) {
     return true;
   }
   return false;

--- a/storage/tianmu/handler/ha_tianmu.cpp
+++ b/storage/tianmu/handler/ha_tianmu.cpp
@@ -1259,8 +1259,10 @@ int ha_tianmu::fill_row(uchar *buf) {
     std::memcpy(buffer.get(), table->record[0], table->s->reclength);
   }
   if (iterator_->IsBase()) {
-    // judge whether this line has been deleted.
-    // if this line has been deleted, data will not be copied.
+    /*
+      Determine whether the current row is invalid in the base layer,
+      If it is invalid, directly return to the SQL layer that the current row has been deleted
+    */
     if (iterator_->BaseCurrentRowIsInvalid()) {
       current_position_ = iterator_->Position();
       iterator_->Next();
@@ -1275,6 +1277,7 @@ int ha_tianmu::fill_row(uchar *buf) {
     if (!delta_record.empty()) {
       switch (core::DeltaRecordHead::GetRecordType(delta_record.data())) {
         case core::RecordType::kInsert:
+          // If the function change fails, it will be considered that the row change has been deleted.
           if (!core::Engine::DecodeInsertRecordToField(delta_record.data(), table->field)) {
             current_position_ = iterator_->Position();
             iterator_->Next();
@@ -1283,6 +1286,11 @@ int ha_tianmu::fill_row(uchar *buf) {
           }
           break;
         case core::RecordType::kUpdate:
+          /*
+            Merge data from the base layer and delta layer.
+            At the same time, record the status of switching to the delta layer
+            When iterating the base layer data, it is judged as invalid for changing rows
+          */
           current_txn_->GetTableByPath(table_name_)->FillRowByRowid(table, iterator_->Position());
           core::Engine::DecodeUpdateRecordToField(delta_record.data(), table->field);
           iterator_->InDeltaUpdateRow.insert(
@@ -1290,6 +1298,11 @@ int ha_tianmu::fill_row(uchar *buf) {
           break;
         case core::RecordType::kDelete:
           current_position_ = iterator_->Position();
+          /*
+            If there is deleted data with a new row in the delta layer,
+            Just record it down,
+            When iterating the base layer data, it is judged as invalid for changing rows
+          */
           iterator_->InDeltaDeletedRow.insert(std::unordered_map<int64_t, bool>::value_type(current_position_, true));
           iterator_->Next();
           dbug_tmp_restore_column_map(table->write_set, org_bitmap);

